GVRf/Framework/src/org/gearvrf/GVRSurfaceView.java

```diff
@@ -0,0 +1,269 @@
+     * with {@link GVRMonoscopicViewManager}
+     * 
+     * @param context
+     *            current context
+     * @param viewManager
+     *            a given {@link GVRMonoscopicViewManager} object to be used in
+     *            {@link GVRSurfaceView}
+     */
+    public GVRSurfaceView(Context context,
+            GVRMonoscopicViewManager viewManager,
+            GVRSurfaceViewRenderer renderer) {
+        super(context);
+        mViewManager = viewManager;
+        /*
+         * To access inputs by onKeyDown().
+         */
+        setFocusable(true);
+        setKeepScreenOn(true);
+        /*
+         * Avoids reloading the application every time it pauses.
+         */
+        setEGLContextClientVersion(3);
+        setPreserveEGLContextOnPause(true);
+        setEGLContextFactory(new GVRContextFactory());
+        setEGLConfigChooser(new GVRConfigChooser(8, 8, 8, 8, 24, 8));
+        if (renderer != null) {
+            renderer.setViewManager(viewManager);
+            setRenderer(renderer);
+        } else {
+            setRenderer(new GVRSurfaceViewRenderer(viewManager));
+        }
+        /*
+         * requestRender() will be called efficiently with VSync.
+         */
+        setRenderMode(RENDERMODE_WHEN_DIRTY);
+    }
+
+    @Override
+    public boolean onKeyDown(int keyCode, KeyEvent event) {
+        mViewManager.onKeyDown(keyCode, event);
+        return super.onKeyDown(keyCode, event);
+    }
+
+    @Override
+    public void doFrame(long frameTimeNanos) {
+        requestRender();
+        Choreographer.getInstance().postFrameCallback(this);
+    }
+
+    @Override
+    public void surfaceCreated(SurfaceHolder holder) {
+        super.surfaceCreated(holder);
+        holder.setFormat(PixelFormat.TRANSLUCENT);
+
+        Choreographer.getInstance().removeFrameCallback(this);
+        Choreographer.getInstance().postFrameCallback(this);
+    }
+
+    @Override
+    public void surfaceChanged(SurfaceHolder holder, int format, int w, int h) {
+        super.surfaceChanged(holder, format, w, h);
+    }
+
+    @Override
+    public void surfaceDestroyed(SurfaceHolder holder) {
+        Choreographer.getInstance().removeFrameCallback(this);
+        super.surfaceDestroyed(holder);
+    }
+}
+
+/*
+ * Copyright (C) 2009 The Android Open Source Project
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+class GVRContextFactory implements GLSurfaceView.EGLContextFactory {
```

Comment:
Perhaps it's too late at night for me, what change are you looking for here?  It's the same header we have on everything else as far as i can tell.